middleware/common/source/domain.cpp

```diff
@@ -129,10 +130,20 @@ namespace casual
 
             file::scoped::Path temp_file{ file::name::unique( path.string(), ".tmp")};
             {
-               std::ofstream output( temp_file);   
+               std::ofstream output( temp_file);
                auto archive = serialize::json::writer();
                archive << model;
                archive.consume( output);
+
+               if ( ! output)
+               {
+                  auto ec =  std::make_error_code( code::system::last::error());
+
+                  if ( std::error_code temp_ec; ! std::filesystem::exists( temp_file, temp_ec))
+                     temp_file.release();
```

Comment:
I don't understand this. If the file does not exist, we release the _temp_file_. Why? Do we get an error in `scoped::Path` dtor otherwise?

Reply:
yes. if the path is non-empty it will try to remove the (non-existing) file. I'll move this check to the destructor for `scoped::Path`  instead